models/models.go

```diff
@@ -347,3 +350,80 @@ func DumpDatabase(filePath string, dbType string) error {
 	}
 	return x.DumpTablesToFile(tbs, filePath)
 }
+
+// DumpDatabaseFixtures dumps all data from database to fixtures files on dirPath
+func DumpDatabaseFixtures(dirPath string) error {
+	for _, t := range tables {
+		if err := dumpTableFixtures(t, dirPath); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func dumpTableFixtures(bean interface{}, dirPath string) error {
+	table := x.TableInfo(bean)
+	f, err := os.Create(filepath.Join(dirPath, table.Name+".yml"))
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+	var bufferSize = 100
+	var objs = make([]interface{}, 0, bufferSize)
+	err = x.BufferSize(bufferSize).Iterate(bean, func(idx int, obj interface{}) error {
+		objs = append(objs, obj)
+		if len(objs) == bufferSize {
+			// BLOCK: need yaml support gonic name mapper
+			data, err := yaml.Marshal(objs)
+			if err != nil {
+				return err
+			}
+			_, err = f.Write(data)
+			if err != nil {
+				return err
+			}
+			objs = make([]interface{}, 0, bufferSize)
+		}
+		return err
+	})
+	if err != nil {
+		return err
+	}
+	if len(objs) > 0 {
+		data, err := yaml.Marshal(objs)
+		if err != nil {
+			return err
+		}
+		_, err = f.Write(data)
+	}
+	return err
+}
+
+// RestoreDatabaseFixtures restores all data from dir to database
+func RestoreDatabaseFixtures(dirPath string) error {
+	for _, t := range tables {
+		if err := restoreTableFixtures(t, dirPath); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func restoreTableFixtures(bean interface{}, dirPath string) error {
+	table := x.TableInfo(bean)
+	data, err := ioutil.ReadFile(filepath.Join(dirPath, table.Name+".yml"))
+	if err != nil {
+		return err
+	}
+
+	var bufferSize = 100
+	v := reflect.MakeSlice(table.Type, 0, bufferSize)
+	// BLOCK: need yaml support gonic name mapper
```

Comment:
There have been 61 commits to `go-yaml/yaml` since it was forked into the `go-gitea` org. Is this still blocked?

Reply:
I will check that.